### PR TITLE
feat: app settings

### PR DESCRIPTION
**Pull request by cpvalente**
Moves all settings to a central place

Screenshot 2023-12-19 at 21 43 21

## Summary by CodeRabbit

- **Documentation**
  - Updated the development guidelines to reflect a specific version requirement for the `node` and `pnpm` dependencies.

### CODE REVIEW

**Comment by cpvalente** [comment]
Very much work in progress, but would you mind taking a look @asharonbaltazar ?

File: apps/client/src/features/app-settings/AppSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+/**
+ * TODO: make store to keep settings
+ *       - whether a tab has unsaved changes
+ */
+
+export default function AppSettings(props: AppSettingsProps) {
+  const { settings = settingPanels[0].id, onClose } = props;
+  useKeyDown(onClose, 'Escape');
+
+  const [selectedTab, setSelectedTab] = useState<SettingsOptionId>(settings);
```
Comment (asharonbaltazar) [inline apps/client/src/features/app-settings/AppSettings.tsx:26]: Why aren't you using the Chakra-UI `Tabs` component?

**Review by asharonbaltazar — COMMENTED** [review]
Left a comment questioning the implementation. What's the advantage to manually writing tabs functionality using a store? Do you want it to sync with the browser params?

File: apps/client/src/features/app-settings/AppSettings.tsx (hunk 1 above)
Comment (cpvalente) [reply apps/client/src/features/app-settings/AppSettings.tsx:26]: We consider abandoning Chakra where possible.

Also the amount of customisation we want would take some effort to do, at which point using the Chakra component no longer provides development speed

**Pull request merged**